lib/postgrex/utils.ex

```diff
@@ -79,11 +79,11 @@ defmodule Postgrex.Utils do
   @spec default_opts(Keyword.t) :: Keyword.t
   def default_opts(opts) do
     opts
-    |> Keyword.put_new(:username, System.get_env("PGUSER") || System.get_env("USER"))
-    |> Keyword.put_new(:password, System.get_env("PGPASSWORD"))
     |> Keyword.put_new(:database, System.get_env("PGDATABASE"))
     |> Keyword.put_new(:hostname, System.get_env("PGHOST") || "localhost")
     |> Keyword.update(:port, normalize_port(System.get_env("PGPORT")), &normalize_port/1)
+    |> Keyword.put_new(:username, System.get_env("PGUSER") || System.get_env("USER"))
```

Comment:
@capitalist Modifying the `default_opts/1` function to something like below should fix the issues that @jmarca is seeing, or at least fix issues that do exist in the current implementation.

```
  def default_opts(opts) do
    password_infering_opts = opts
    |> Keyword.put_new(:database, System.get_env("PGDATABASE"))
    |> Keyword.put_new(:hostname, System.get_env("PGHOST") || "localhost")
    |> Keyword.update(:port, normalize_port(System.get_env("PGPORT")), &normalize_port/1)
    |> Keyword.put_new(:username, System.get_env("PGUSER") || System.get_env("USER"))
    |> Keyword.put_new(:pgpassfile, System.get_env("PGPASSFILE"))
    |> Keyword.put_new_lazy(:pgpassfile, &Postgrex.Pgpass.os_default_location/0)

    password_infering_opts
    |> Keyword.put_new_lazy(:password, fn -> System.get_env("PGPASSWORD") || Postgrex.Pgpass.password(password_infering_opts) end)
    |> Keyword.put_new(:types, Postgrex.DefaultTypes)
    |> Enum.reject(fn {_k, v} -> is_nil(v) end)
  end
```